dhall/Mock/package.dhall

```diff
@@ -1,23 +1,67 @@
-let HttpMethod = < HEAD | PUT | POST | GET | DELETE >
+let HttpMethod = < CONNECT | DELETE | GET | HEAD | OPTIONS | PATCH | POST | PUT | TRACE >
 
-let HttpRequest : Type
-    = { method  : Optional HttpMethod
-      , path    : Optional Text
+let Body = < JSON : { json : Text } | TEXT : { text : Text } >
+
+let QueryParam = { key: Text, value: Text }
+
+let Header = { mapKey: Text, mapValue: Text }
+
+let HttpRequest
+    = { Type = { method  : Optional HttpMethod
+               , path    : Optional Text
+               , body    : Optional Body
+               , params  : List QueryParam
+               , headers : List Header
+               }
+      , default = { method  = None HttpMethod
+                  , path    = None Text
+                  , body    = None Body
+                  , params  = [] : List QueryParam
+                  , headers = [] : List Header
+                  }
       }
 
-let HttpResponse : Type =
-      { statusCode   : Optional Integer
-      , statusReason : Optional Text
-      , body         : Optional Text
+let HttpResponse 
+    = { Type = { statusCode   : Optional Natural
+               , statusReason : Optional Text
+               , body         : Optional Text
+               , headers      : List Header
+               }
+      , default = { statusCode   = None Natural
+                  , statusReason = None Text
+                  , body         = None Text
+                  , headers      = [] : List Header
+                  }
       }
 
 let Expectation : Type =
-      { request  : HttpRequest
-      , response : HttpResponse
+      { request  : HttpRequest.Type
+      , response : HttpResponse.Type
       }
 
-in { HttpMethod   = HttpMethod
-   , HttpRequest  = HttpRequest
-   , HttpResponse = HttpResponse
-   , Expectation  = Expectation
+let contentTypeJSON : Header = 
+  { mapKey = "Content-Type", mapValue = "application/json" }
+
+let contentTypeXML : Header = 
+  { mapKey = "Content-Type", mapValue = "application/xml" }
+
+let contentTypeText : Header = 
+  { mapKey = "Content-Type", mapValue = "text/plain"}
+
+in { HttpMethod      = HttpMethod
+   , QueryParam      = QueryParam
+   , Header          = Header
+   , Body            = Body
+   , HttpRequest     = HttpRequest
+   , HttpResponse    = HttpResponse
+   , Expectation     = Expectation
+   , statusOK            = Some 200
```

Comment:
Could we add all the status in a custom object ?
Something like 
```
Status = {
  OK = Some 200
  , Created = Some 201 
  , ...
}
```

Reply:
good idea

Reply:
In convention in dhall, only Type start with an uppercase, but we can get rid of this. What do you think?

Reply:
It's also shorter to define status by :

```
Mock.statusOK
```

than : 
```
Mock.Status.OK
```

Reply:
OK Tas more thinking about doing an import on Status direct but fine with me